packages/web-app-files/src/composables/parentShare/useIncomingParentShare.ts

```diff
@@ -0,0 +1,34 @@
+import { buildShare } from '../../helpers/resources'
+import { useStore } from 'web-pkg/src/composables'
+import { computed, ref, unref } from '@vue/composition-api'
+import { useTask } from 'vue-concurrency'
+import { clientService } from 'web-pkg/src/services'
+
+export function useIncomingParentShare() {
+  const store = useStore()
+  const incomingParentShare = ref(null)
+  const sharesTree = computed(() => store.state.Files.sharesTree)
+
+  const loadIncomingParentShare = useTask(function* (signal, resource) {
+    let parentShare
+    for (const shares of Object.values(unref(sharesTree)) as any) {
+      parentShare = shares.find((s) => s.incoming)
+      if (parentShare) {
+        incomingParentShare.value = parentShare
+        return
+      }
+    }
+
+    if (resource.shareId) {
+      const parentShare = yield clientService.owncloudSdk.shares.getShare(resource.shareId)
```

Comment:
the `parentShare` variable is already defined above, so you should drop the `const` here, no?